### PR TITLE
Add guidelines for templating images in helm charts

### PR DESCRIPTION
**Pull request by rossf7**
Prepares us for using App Catalog and reducing some of the image pulling pain we see in AWS China.

### CODE REVIEW

File: helm/image_templating.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+## Templating
+
+When templating container images in charts `values.yaml` should be structured
+like this. This allows us to override `image.registry` in some control planes
+e.g. AWS China.
+
+
+```yaml
+image:
+  registry: quay.io
```
Comment (tomahawk28) [inline helm/image_templating.md]: Not related to this change but most of the charts are simply `quay.io` images, can we may set up the default value for that?

**Review by sslavic — APPROVED** [review]
LGTM

File: helm/image_templating.md (hunk 1 above)
Comment (rossf7) [reply helm/image_templating.md]: Yes we use `quay.io` everywhere except AWS China where we use `registry-intl.cn-shanghai.aliyuncs.com`.

But at some point we may want to run our own registry like Harbor and for some onprem control planes as well as China.

**Review by tomahawk28 — APPROVED** [review]

**Comment by rossf7** [comment]
Fishing for 1 more approval since this is `fmt`.

**Review by tfussell — APPROVED** [review]
Makes sense to me!

**Pull request merged**